osu.Game/Screens/Edit/Compose/Components/Timeline/TimelineHitObjectBlueprint.cs

```diff
@@ -275,32 +275,33 @@ protected override void OnDrag(DragEvent e)
 
                 OnDragHandled?.Invoke(e);
 
-                var time = timeline.GetTimeFromScreenSpacePosition(e.ScreenSpaceMousePosition);
-
-                switch (hitObject)
+                if (timeline.SnapScreenSpacePositionToValidTime(e.ScreenSpaceMousePosition).Time is double time)
```

Comment:
Not sure if related to the above slider issues, but there's some weird interaction here too https://drive.google.com/file/d/1xDEzIL0_M5QLu9gYFfyje0N_wKCoJ444

Reply:
I can't seem to make this happen. Does it not happen on master? Can you link the beatmap?

Reply:
https://osu.ppy.sh/beatmapsets/1145924#osu/2392543 master behaves a little better. On master a repeat is added as soon as you cross a tick, whereas here it seems you have to cross the next mid-point for the same to happen.

Reply:
Excellent, should be fixed (value was being floored instead of rounded, relying on no snapping previously).